hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2160,3 +2155,65 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
         return numpy.median(self._data)
 
     return hpat_pandas_series_median_impl
+
+@overload_method(SeriesType, 'pct_change')
+def hpat_pandas_series_pct_change(self, periods=1, fill_method='pad', limit=None, freq=None):
+    """
+    Pandas Series method :meth:`pandas.Series.pct_change` implementation.
+
+    Note: Unsupported mixed numeric and string data
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_pct_change
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_pct_change2
+
+    Parameters
+    -----------
+    self: :obj:`pandas.Series`
+        input series
+    periods: :obj:`int`, default 1
+    fill_method: :obj:`str`, default 'pad'
+    limit: :obj:`int`, default None
+    freq: :obj: DateOffset, timedelta, or offset alias string, optional
+            Increment to use from time series API (e.g. 'M' or BDay()).
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` object
+    """
+
+    _func_name = 'Method pct_change().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(self.dtype, types.Number):
```

Comment:
Please use `self.data.dtype` instead of `self.dtype` that looks more explicitly. Analogically you need to check types of all parameters and cover such checks by unit tests.

Reply:
> `dtype`

Please make the change in all places.